std/haxe/macro/Expr.hx

```diff
@@ -169,154 +213,603 @@ enum Unop {
 	OpNegBits;
 }
 
+/**
+	Represents a AST node for the AST.
+	@see http://haxe.org/manual/macro-reification-expression.html
+**/
 typedef Expr = {
+	/**
+		The expression kind.
+	**/
 	var expr : ExprDef;
+	
+	/**
+		The position of the expression.
+	**/
 	var pos : Position;
 }
 
+/**
+	Represents a AST node identical to `Expr`, but it allows constraining the 
+	type of accepted expressions.
+	@see http://haxe.org/manual/macro-ExprOf.html
+**/
 typedef ExprOf<T> = Expr;
 
+/**
+	Represents a case for the AST.
+	@see http://haxe.org/manual/expression-switch.html
+**/
 typedef Case = {
+	/**
+		The value expressions of the case.
+	**/
 	var values : Array<Expr>;
+	/**
+		The guard expressions of the case. Guards restrict cases.
+	**/
 	@:optional var guard : Null<Expr>;
+	/**
+		The expression of the case.
+	**/
 	var expr: Null<Expr>;
 }
 
+/**
+	Represents a variable for the AST.
+	@see http://haxe.org/manual/expression-var.html
+**/
 typedef Var = {
+	/**
+		The name of the variable.
+	**/
 	name : String,
+	/**
+		The type of the variable.
+	**/
 	type : Null<ComplexType>,
+	/**
+		The expression of the variable.
+	**/
 	expr : Null<Expr>
 }
 
+/**
+	Represents a catch for the AST.
+	@http://haxe.org/manual/expression-try-catch.html
+**/
 typedef Catch = {
+	/**
+		The name of the catch.
+	**/
 	name : String,
+	/**
+		The type of the catch.
+	**/
 	type : ComplexType,
+	/**
+		The expression of the catch.
+	**/
 	expr : Expr
 }
 
+/**
+	Represents kind of a node for the AST.
```

Comment:
I don't know if there's an article missing here but "kind of a node" sounds really stupid. :D